network/msgCompressor.go

```diff
@@ -126,18 +110,16 @@ func (dec zstdProposalDecompressor) convert(data []byte) ([]byte, error) {
 
 func (c *wsPeerMsgDataConverter) convert(tag protocol.Tag, data []byte) ([]byte, error) {
 	if tag == protocol.ProposalPayloadTag {
-		if c.ppdec.enabled() {
-			// sender might support compressed payload but fail to compress for whatever reason,
-			// in this case it sends non-compressed payload - the receiver decompress only if it is compressed.
-			if c.ppdec.accept(data) {
-				res, err := c.ppdec.convert(data)
-				if err != nil {
-					return nil, fmt.Errorf("peer %s: %w", c.origin, err)
-				}
-				return res, nil
+		// sender might support compressed payload but fail to compress for whatever reason,
+		// in this case it sends non-compressed payload - the receiver decompress only if it is compressed.
+		if c.ppdec.accept(data) {
+			res, err := c.ppdec.convert(data)
+			if err != nil {
```

Comment:
Can this fail for any reason other than not being compressed?

Reply:
It fails on Read errors and too zip bombs